src/Bicep.Core/Registry/PublicRegistry/PublicRegistryModuleMetadataProvider.cs

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
```

Comment:
Most of this was just moved, but i did make some changes to how the cache is updaeted